### PR TITLE
Binding Generator

### PR DESCRIPTION
**Pull request by arrudagates**

### CODE REVIEW

**Comment by arrudagates** [comment]
Requires [this PR](https://github.com/sensorial-systems/ligen/pull/45) due to changes introduced in it.

File: core/src/ast/typ.rs [hunk 1]
```diff
@@ -70,6 +72,81 @@ impl Type {
...
+                | Integer::U32
+                | Integer::U64
+                | Integer::U128
+                | Integer::USize => Atomic::UnsignedInt,
+                Integer::I8
+                | Integer::I16
+                | Integer::I32
+                | Integer::I64
+                | Integer::I128
+                | Integer::ISize => Atomic::Int,
```
Comment (notdanilo) [inline core/src/ast/typ.rs:90]: Fix this

File: core/src/ast/typ.rs [hunk 2]
```diff
@@ -70,6 +72,81 @@ impl Type {
...
+            },
+            ligen_core::ir::Atomic::Float(_) => Atomic::Float,
```
Comment (notdanilo) [inline core/src/ast/typ.rs:92]: And this. Must be translated to `float` and `double`

File: core/src/ast/typ.rs [hunk 3]
```diff
@@ -70,6 +72,81 @@ impl Type {
...
+            ligen_core::ir::Atomic::Boolean => panic!("Boolean not implemented"),
+            ligen_core::ir::Atomic::Character => Atomic::Char,
```
Comment (notdanilo) [inline core/src/ast/typ.rs:94]: char in Rust is 16bit, while in c is 8bit. It won't be compatible

File: core/src/.#proc_macro.rs [hunk 4]
```diff
@@ -0,0 +1 @@
+arruda@manjaro.11446:1621335731
```
Comment (notdanilo) [inline core/src/.#proc_macro.rs:1]: What's this file?

**Comment by notdanilo** [comment]
This PR was continued in another branch

**Pull request closed without merging**